src/site/content/en/blog/detached-window-memory-leaks/index.md

```diff
@@ -0,0 +1,550 @@
+  iframe persisting after other references have been cleaned up.
+
+- A memory-heavy document loaded in a window or iframe can be accidentally kept in-memory long
+  after navigating to a new URL. This is often caused by the parent page retaining references to
+  the document in order to allow for listener removal.
+
+- When passing a JavaScript object to another window or iframe, the Object's prototype chain
+  includes references to the environment it was created in, including the window that created it.
+  This means it's just as important to avoid holding references to objects from other windows as
+  it is to avoid holding references to the windows themselves.
+  <table class="w-table--top-align"><tbody><tr>
+  <td>
+
+{% Label %}index.html:{% endLabel %}
+
+```html
+<script>
+  let currentFiles;
+  function load(files) {
+    // this retains the popup:
+    currentFiles = files;
+  }
+  window.open('upload.html');
+</script>
+```
+
+  </td>
+  <td>
+
+{% Label %}upload.html:{% endLabel %}
+
+```html
+<input type="file" id="file" />
+<script>
+  file.onchange = () => {
+    parent.load(file.files);
+  };
+</script>
+```
+
+  </td>
+  </tr></tbody></table>
+
+## Detecting memory leaks caused by detached windows {: #detect-leaks }
+
+Tracking down memory leaks can be tricky. It is often difficult to construct isolated reproductions
+of these issues, particularly when multiple documents or windows are involved. To make things more
+complicated, inspecting potential leaked references can end up creating additional references that
+prevent the inspected objects from being garbage collected. To that end, it's useful to start with
+tools that specifically avoid introducing this possibility.
+
+A great place to start debugging memory problems is to
+[take a heap snapshot](https://developers.google.com/web/tools/chrome-devtools/memory-problems#discover_detached_dom_tree_memory_leaks_with_heap_snapshots).
+This provides a point-in-time view into the memory currently used by an application - all the
+objects that have been created but not yet garbage-collected. Heap snapshots contain useful
+information about objects, including their size and a list of the variables and closures that
+reference them.
+
+<figure class="w-figure">
+  <img class="w-screenshot" src="./heap-snapshot.png" width="762"
+       alt="A screenshot of a heap snapshot in Chrome DevTools showing the references that retain
+       a large object.">
+  <figcaption class="w-figcaption">
+    A heap snapshot showing the references that retain a large object.
+  </figcaption>
+</figure>
+
+To record a heap snapshot, head over to the **Memory** tab in Chrome DevTools and select **Heap
+Snapshot** in the list of available profiling types. Once the recording has finished, the
+**Summary** view shows current objects in-memory, grouped by constructor.
+
+<figure class="w-figure">
+  <video controls autoplay loop muted playsinline width="640">
```

Comment:
```suggestion
  <video controls autoplay loop muted playsinline width="640" class="w-screenshot">
```
Adds a little grey border around the video which should make it look a bit more polished